addons/account/models/account_invoice.py

```diff
@@ -1360,7 +1360,7 @@ def _prepare_payment_vals(self, pay_journal, pay_amount=None, date=None, writeof
             'amount': pay_amount or self.residual,
             'payment_date': date or fields.Date.context_today(self),
             'communication': communication,
-            'partner_id': self.partner_id.id,
+            'partner_id': self.partner_id.commercial_partner_id.id,
```

Comment:
I don't think we want to change such a behavior in stable... This was introduced back in v9 in 2015.